Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/springcloud/SpringCloudAppNode.java

```diff
@@ -190,148 +193,44 @@ private void fillData(AppResourceInner newApp, DeploymentResourceInner deploy) {
-        }
-
-        @Override
-        public String getProgressMessage() {
-            return Node.getProgressMessage(AzureActionEnum.OPEN_IN_PORTAL.getDoingName(), SpringCloudModule.MODULE_NAME, SpringCloudAppNode.this.name);
-        }
-
-        @Override
-        public TelemetryParameter getTelemetryParameter() {
-            return TelemetryParameter.SpringCloud.OPEN_IN_PORTAL;
-        }
+    private void openInPortal() {
+        SpringCloudAppNode.this.openResourcesInPortal(SpringCloudAppNode.this.getSubscriptionId(), SpringCloudAppNode.this.getAppId());
     }
 
-    // Show Properties
-    private class ShowPropertiesAction extends NodeActionListener implements ActionTelemetrable {
-
-        @Override
-        protected void actionPerformed(NodeActionEvent e) {
-            DefaultLoader.getUIHelper().openSpringCloudAppPropertyView(SpringCloudAppNode.this);
-            // add this statement for false updating notice to update Property view
-            // immediately
-            SpringCloudStateManager.INSTANCE.notifySpringAppUpdate(SpringCloudAppNode.this.clusterId,
-                    SpringCloudAppNode.this.app, SpringCloudAppNode.this.deploy);
-        }
-
-        @Override
-        public TelemetryParameter getTelemetryParameter() {
-            return TelemetryParameter.SpringCloud.SHOW_PROPERTIES;
-        }
+    private void showProperties() {
+        DefaultLoader.getUIHelper().openSpringCloudAppPropertyView(SpringCloudAppNode.this);
+        // add this statement for false updating notice to update Property view
+        // immediately
+        SpringCloudStateManager.INSTANCE.notifySpringAppUpdate(SpringCloudAppNode.this.clusterId,
+                SpringCloudAppNode.this.app, SpringCloudAppNode.this.deploy);
     }
 
-    // Open in browser action class
-    private class OpenInBrowserAction extends NodeActionListener implements ActionBackgroundable, ActionTelemetrable {
-
-        @Override
-        protected void actionPerformed(NodeActionEvent e) {
-            if (StringUtils.isNotEmpty(SpringCloudAppNode.this.app.properties().url())) {
-                DefaultLoader.getUIHelper().openInBrowser(SpringCloudAppNode.this.app.properties().url());
-            } else {
-                DefaultLoader.getUIHelper().showInfo(SpringCloudAppNode.this, "Public url is not available for app: " + app.name());
-            }
-        }
-
-        @Override
-        public String getProgressMessage() {
-            return Node.getProgressMessage("Opening", SpringCloudModule.MODULE_NAME, SpringCloudAppNode.this.name);
-        }
-
-        @Override
-        public TelemetryParameter getTelemetryParameter() {
-            return TelemetryParameter.SpringCloud.OPEN_IN_BROWSER;
+    private void openInBrowser() {
+        if (StringUtils.isNotEmpty(SpringCloudAppNode.this.app.properties().url())) {
```

Comment:
Same as above, no need to call like a inner class

Reply:
updated in the next PR.